kubernetes/annotations_and_labels.md

```diff
@@ -41,25 +39,46 @@ This page defines common annotations and labels we set in Kubernetes objects.
 - `release.giantswarm.io/version` - value should be Giant Swarm release
   version, e.g. `release.giantswarm.io/version=2.3.0`.
 - `OPERATOR.giantswarm.io/version` - value should be the version of the
-  operator as defined in `project.go` and matching `appVersion` in
-  `Chart.yaml`, e.g. `kvm-operator.giantswarm.io/version=1.0.0`. It is used by
-  the given operator to recognize which object it should reconcile (i.e. to
-  only reconcile objects matching its own version). When set on Node objects it
-  is used to set that information in the status with the statusresource. This
-  is different from release version and can be the same in multiple releases.
-  Its value may be equal to that of `app.kubernetes.io/version` but it has a
-  different purpose and since there could be multiple operators reconciling one
-  object there could be multiple per-operator labels on one object.
-- `app.giantswarm.io/branch` - (informational) branch from which this
-  instance of the app/operator was built from.
-- `app.giantswarm.io/commit` - (informational) ID (git SHA) of the commit from
-  which this instance of the app/operator was built from.
+  operator reconciling the object, as defined in `project.go` and matching
+  `appVersion` in `Chart.yaml`, e.g.
+  `kvm-operator.giantswarm.io/version=1.0.0`. It is used by the given operator
+  to recognize which object it should reconcile (i.e. to only reconcile objects
+  matching its own version). When set on Node objects it is used to set that
+  information in the status with the statusresource. This is different from
+  release version and can be the same in multiple releases. Its value may be
+  equal to that of `app.kubernetes.io/version` but it has a different purpose
+  and since there could be multiple operators reconciling one object there
+  could be multiple per-operator labels on one object.
+- `helm.sh/chart` - value should contain a chart name and version, see
+  [Helm chart labels best practices][helm-labels]
+- `app.kubernetes.io/name` - the name of the application; should be applied to
+  every Kubernetes resource associated with that application, i.e. all objects
+  in a chart that installs it. E.g. `app.kubernetes.io/name=kvm-operator`.
+- `app.kubernetes.io/instance` - value should be set to `{{ .Release.Name }}`
+  and is meant for differentiating between instances of the same application.
+  This together with `app.kubernetes.io/name` should be used as replicas
+  selector, e.g.
+  `app.kubernetes.io/name=kvm-operator,app.kubernetes.io/instance=kvm-operator-1.0.0`.
+- informational labels, applied to all objects in a chart that installs an app
+  or operator and so they indicate its source and instance:
+  - `app.kubernetes.io/version` - value should be the app/operator version as
```

Comment:
Just a thought: since https://github.com/giantswarm/gatekeeper-control-plane-rules/pull/20, `app.kubernetes.io/version` is used for gatekeeper uniqueness checks [(see here)](https://github.com/giantswarm/gatekeeper-control-plane-rules/blob/master/helm/gatekeeper-control-plane-rules/templates/uniqueoperatorversion.yaml#L16).

It would be helpful to include this information here, somewhere.

Reply:
Em, that's not right is it? I mean the operators still use the `$OPERATOR.giantswarm.io/version` label for marking objects they reconcile, e.g.:

- https://github.com/giantswarm/aws-operator/blob/master/service/controller/cluster_resource_set.go#L719
- https://github.com/giantswarm/aws-operator/blob/ad94f07f4a83ff2cb31b786deddf9ce64af92197/service/controller/key/common.go#L122
- https://github.com/giantswarm/aws-operator/blob/ad94f07f4a83ff2cb31b786deddf9ce64af92197/pkg/label/label.go#L18

I missed this change in GK rules, why was that done?

Reply:
GK rules are to make sure we don't install two deployments with the same `app.kubernetes.io/name` and `app.kubernetes.io/version`.

BTW @wtty-fool we should also rename `app` to `app.kubernetes.io/name` here https://github.com/giantswarm/gatekeeper-control-plane-rules/pull/20/files#diff-d639fe45fa938f521566f27263b98d82R17

Reply:
Yeah, oh right so this is because we removed `$OPERATOR.giantswarm.io/version` from the operator's own `Deployment` object. Meaning it now compares the internal project version which is exposed as `app.kubernetes.io/version` label on its own components, to `$OPERATOR.giantswarm.io/version` set on reconciled objects. OK, makes sense, I'll update descriptions.

Reply:
Done. I'm going to leave this until tomorrow so in case the wording can be improved let me know.